### ISSUE TITLE
Window of figure can't be deleted after creating a figure

### ISSUE THREAD

**Issue by ildoildi** [matplotlib/matplotlib#15398]
### Bug report
When I create a figure then try to clear figure the action is completed but window after the figure remain opened and move around to other canvas making impossible to run the script normally, windows remained opened one by other. I use fig.clf() and it works but does not close the window created on tkinter canvas
**Bug summary**
How to close window created together with figure

**Code for reproduction**
******code for screenshot with figure*****
```python
def bow():
        fig = Figure(figsize=(6,3), dpi=120)
        ax = fig.add_axes([0.02, 0.02, 0.95, 0.95], 
                           projection=alfa.PlateCarree())
        ax.set_global()
        ax.stock_img()
        ax.coastlines()
        ax.gridlines()
        canvas3 = FigureCanvasTkAgg(fig, master=ildi)
        canvas3.draw()
        canvas3.get_tk_widget().pack(side=TOP, expand=1)
        return bow
btntest1 = Button(toolbar1, text= "test 2",compound=LEFT, activebackground="bisque1", command=bow)
btntest1.pack(side=LEFT, padx=2, pady=2)
 ****** code for empty window screenshot******
def bow():

        fig = Figure(figsize=(6,3), dpi=120)
        ax = fig.add_axes([0.02, 0.02, 0.95, 0.95], 
                           projection=alfa.PlateCarree())
        ax.set_global()
        ax.stock_img()
        ax.coastlines()
        ax.gridlines()
        fig.clf()
        canvas3 = FigureCanvasTkAgg(fig, master=ildi)
        canvas3.draw()
        canvas3.get_tk_widget().pack(side=TOP, expand=1)
        return bow
```

```python
# Paste your code here
#
def bow():

        fig = Figure(figsize=(6,3), dpi=120)

        ax = fig.add_axes([0.02, 0.02, 0.95, 0.95], 
                           projection=alfa.PlateCarree())
        
        ax.set_global()
        ax.stock_img()
        ax.coastlines()
        ax.gridlines()
        fig.clf()
        canvas3 = FigureCanvasTkAgg(fig, master=ildi)
        canvas3.draw()
        canvas3.get_tk_widget().pack(side=TOP, expand=1)
        return bow
#
```

**Actual outcome**

![Capture](https://user-images.githubusercontent.com/56173748/66517650-2b3b5180-eaec-11e9-8c68-bbf7f28a8d81.PNG)
![Capture1](https://user-images.githubusercontent.com/56173748/66517669-34c4b980-eaec-11e9-9234-1b73a82a37ce.PNG)

```
# If applicable, paste the console output here
#
#
```

**Expected outcome**

**Matplotlib version**

  * Operating system: Windows 10
  * Matplotlib version: 3.1.1
  * Matplotlib backend (`print(matplotlib.get_backend())`):
  * Python version: 3.7
  * Jupyter version (if applicable):
  * Other libraries: 

[@tacaswell edited to add markup]

**Comment by tacaswell** [comment]
Are you running the Tk main loop someplace?

There are a number of un-defined variables in your example, can you reproduce the problem with code we can copy-paste to run?

**Comment by ildoildi** [comment]
`
#libraries

from tkinter import*
from matplotlib import*
from cartopy import*
from tkinter import ttk
from tkinter import filedialog
from tkinter import messagebox
import cartopy.crs as alfa
import cartopy.feature as cfeature
import matplotlib.pyplot as amarildo
import requests, os, sys, zipfile, shutil, subprocess
import numpy
import pandas
import geopandas as gpd
import shapefile as shp
import seaborn as beta
from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg
from matplotlib.figure import Figure

#definitions
def bow():

        fig = Figure(figsize=(6,3), dpi=120)

        ax = fig.add_axes([0.02, 0.02, 0.95, 0.95], 
                           projection=alfa.PlateCarree())
        
        ax.set_global()
        ax.stock_img()
        ax.coastlines()
        ax.gridlines()

        #fig.clf()

        canvas3 = FigureCanvasTkAgg(fig, master=ildi)
        canvas3.draw()
        canvas3.get_tk_widget().pack(side=TOP,expand=1)
        return bow
#main unit of program
ildi = Tk()
ildi.title("AutoCord 0.1.7")
ildi.geometry("1200x650")
ildi.configure(bg="darkslategray2")
ildi.resizable(0,0)

#toolbar
toolbar1=Frame(ildi, bg="azure2")
btntest = Button(toolbar1, text= "test 1",compound=LEFT, activebackground="bisque1", command=arow)
btntest.pack(side=LEFT, padx=2, pady=2)
btntest1 = Button(toolbar1, text= "test 2",compound=LEFT, activebackground="bisque1", command=bow)
btntest1.pack(side=LEFT, padx=2, pady=2)

toolbar1.tk_focusFollowsMouse()
toolbar1.pack(side=TOP, fill=X)

#Canvas and lines
canvas1 = Canvas(ildi, width=300, height=614, bg="white")
canvas1.pack(side=LEFT)

canvas2 = Canvas(ildi, width=888, height=100, bg="white")
canvas2.place(x=305,y=31)

canvas3 = Canvas(ildi, width=888, height=508, bg="white")
canvas3.place(x=305,y=136)

#close the code
ildi.mainloop()




`

**Comment by ildoildi** [comment]
this is full code related to the problem, again I'm repeating the problem is that the window created on canvas can't be deleted. fig.clf() function delete axis and figure but a blank white window is remained opened in canvas and when I click again to rewrite the figure all what I get is a bug where the last figure is moved up and a second one is created, for visual view of this check the screenshots I attached to this issue

**Comment by tacaswell** [comment]
Can you simplify it to not use any domain specific libraries (geopandas, cartopy, pandas, etc)?

I do not understand what this is supposed to do or what you are expecting.  `fig.clf()` clears the figure, but does not do anything to destroy the underlying canvas, for that you would have to remove / destroy they tk widget your self.

This may be better discussed at https://discourse.matplotlib.org/c/community

**Comment by ildoildi** [comment]
After 36 hours of testing I found the solution of using Matplotlib to clear and replace Canvas created with Tkinter. So to short explain I just put matplotlib in role of a "menager" for data created with Cartopy and visualised on Tkinter. It's amaizing because now I can use it like 3 in 1, and the result is much more better. And yes fig.clf() can clear window created by other library(somehow).  I'm closing the issue cuz I found the solution.
